mantis-control-plane/mantis-control-plane-server/src/main/java/io/mantisrx/master/jobcluster/job/JobActor.java

```diff
@@ -2269,7 +2274,7 @@ private void resubmitWorker(JobWorker oldWorker) throws Exception {
         @Override
         public int scaleStage(MantisStageMetadataImpl stageMetaData, int numWorkers, String reason) {
             LOGGER.info("Scaling stage {} to {} workers", stageMetaData.getStageNum(), numWorkers);
-            final int oldNumWorkers = stageMetaData.getNumWorkers();
+            final int oldNumWorkers = stageMetaData.getAllWorkers().size();
```

Comment:
Can you check when the worker list has out of sync workers (e.g. worker index 5 when numWorker is 3) such workers will get closed?